service/view_building_coordinator.cc

```diff
@@ -432,6 +435,51 @@ future<canonical_mutation> view_building_coordinator::remove_built_view(const gr
     co_return canonical_mutation(std::move(mut));
 }
 
+static bool contains_range(const dht::token_range_vector& ranges, const dht::token_range& range) {
+    return std::find(ranges.cbegin(), ranges.cend(), range) != ranges.cend();
+}
+
+future<std::vector<mutation>> view_building_coordinator::get_migrate_tasks_mutations(const group0_guard& guard, table_id table_id, std::optional<locator::tablet_replica> abandoning_replica, std::optional<locator::tablet_replica> pending_replica, dht::token_range range) {
+    auto state = co_await load_coordinator_state();
```

Comment:
You are calling `load_coordinator_state` in multiple places, on multiple occasions. I think that it is potentially wasteful and can result in it being loaded more times than necessary. It should be called in `group0_state_machine::load_snapshot`/`group0_state_machine::apply`, like we do for the topology coordinator.

We can leave it for a follow-up, though.